components/eam/src/chemistry/mozart/lin_strat_chem.F90

```diff
@@ -277,7 +277,7 @@ subroutine linv3_strat_chem_solve(ncol, lchnk, xvmr, h2ovmr, xsfc, o3col,temp, s
     real(r8), intent(in)                           :: delta_t             ! timestep size (secs)
     real(r8), intent(in)                           :: rlats(ncol)         ! column latitudes (radians)
     integer,  intent(in)   , dimension(pcols)      :: ltrop               ! chunk index    
-    real(r8), intent(in)   , dimension(ncol ,pver) :: pdeldry             !  dry pressure delta about midpoints (Pa) 
+    real(r8), intent(in)   , dimension(pcols,pver) :: pdeldry             !  dry pressure delta about midpoints (Pa) 
```

Comment:
Qi, it doesn't need to be as long as they are referenced properly in calculations. Sure, making the dimension the same will make the codes less prone to mistake. 

In the current case, rlats is later used as a whole array (as in 'lats = rlats * radians_to_degrees') without explicit scoping, so it happens to avoid debug mode runtime floating point issue (say, undefined value, NaN of Infinity being used in multiplication). Pdeldry is different in the calculations as explicit scoping (1:ncol) is always used. It also helps rlats is a one-dimension array. Even if the caller passes in rlats of size pcols with undefined value beyond ncol, it wouldn't cause any problem.

Alternatively for pdeldry, change the caller to pass pdeldry with explicit scoping, namely pdeldry(:ncol,:), then no changes needed for the declaration of pdeldry in lin_strat_chem.F90. The floating invalid was detected inside lin_strat_chem.F90 and I was tracing which variable in that file containing the invalid value , so I chose to make change there.